android/app/src/main/java/org/mozilla/magnet/api/Api.java

```diff
@@ -0,0 +1,148 @@
+    }
+
+    protected Store getCache() {
+        return Store.get(mContext);
+    }
+
+    public void post(String path, HashMap<String,Object> data, Callback callback) {
+        Api match = find(path);
+        if (match != null) match.post(path, data, callback);
+    }
+
+//    public void put(String path) {
+//        Api match = find(path);
+//        if (match != null) match.get(path);
+//    }
+
+    public void delete(String path, HashMap<String,Object> data, Callback callback) {
+        Api match = find(path);
+        if (match != null) match.delete(path, data, callback);
+    }
+
+    private RequestQueue getQueue() {
+        if (mQueue != null) return mQueue;
+        mQueue = Volley.newRequestQueue(mContext);
+        return mQueue;
+    }
+
+    protected void requestJsonArray(String url, final Callback callback) {
+        request(Request.Method.GET, url, null, new Callback() {
+            @Override
+            public void callback(String error, Object result) {
+                if (error != null) {
+                    callback.callback(error, null);
+                    return;
+                }
+
+                try {
+                    callback.callback(null, new JSONArray((String) result));
+                } catch (JSONException err) {
+                    callback.callback(err.getMessage(), null);
+                }
+
+            }
+        });
+    }
+
+    protected void requestJsonObject(String url, final Callback callback) {
+        request(Request.Method.GET, url, null, new Callback() {
+            @Override
+            public void callback(String error, Object result) {
+                if (error != null) {
+                    callback.callback(error, null);
+                    return;
+                }
+
+                try {
+                    callback.callback(null, new JSONObject((String) result));
+                } catch (JSONException err) {
+                    callback.callback(err.getMessage(), null);
+                }
+
+            }
+        });
+    }
+
+    private void request(int method, String url, HashMap data, final Callback callback) {
+        StringRequest request = new StringRequest(method, url, new Response.Listener<String>() {
+            @Override
+            public void onResponse(String response) {
+                callback.callback(null, response);
+            }
+        }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
+                callback.callback(error.getMessage(), null);
+            }
+        });
+
+        getQueue().add(request);
+    }
+
+    protected boolean hasConnectivity() {
+        ConnectivityManager cm = (ConnectivityManager) mContext.getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
+        return activeNetwork != null && activeNetwork.isConnectedOrConnecting();
+    }
+
+    public interface Callback {
```

Comment:
can we change this interface to define two methods, one for success and one for error, kind or promise reject and resolve.